compiler/literal/src/complex.rs

```diff
@@ -0,0 +1,73 @@
+use crate::float;
+
+/// Convert a complex number to a string.
+pub fn to_string(re: f64, im: f64) -> String {
+    // integer => drop ., fractional => float_ops
+    let mut im_part = if im.fract() == 0.0 {
+        im.to_string()
+    } else {
+        float::to_string(im)
+    };
+    im_part.push('j');
+
+    // positive empty => return im_part, integer => drop ., fractional => float_ops
+    let re_part = if re == 0.0 {
+        if re.is_sign_positive() {
+            return im_part;
+        } else {
+            re.to_string()
```

Comment:
```suggestion
            "0"
```
or
```suggestion
            "0.0"
```
?

Calling specifically `f64::to_string()` only for zero and `float::to_string(re)` for non-zero looks like a bug or unnecessary

Reply:
This was copied over from the old implementation, but I agree.

Reply:
oh, sorry!